src/resolver.rs

```diff
@@ -143,22 +136,14 @@ impl Resolver {
 
         base_document.now(now);
         let mut last_updated_did_version_id: Option<U64> = None;
-        let mut deactivated = false;
 
         for (event, meta) in history {
             let LogMeta { block_number, .. } = meta;
 
             if version_id.unwrap_or_default() > U64::zero() {
                 if meta.block_number <= version_id.unwrap_or_default() {
                     // 1. delegate events
-                    Resolver::dispatch_event(
-                        self,
-                        &mut base_document,
-                        public_key,
-                        event,
-                        meta,
-                        &mut deactivated,
-                    );
+                    Resolver::dispatch_event(self, &mut base_document, public_key, event, meta);
                     // 2. set latest version
                     if current_version_id < block_number {
```

Comment:
I suggest also considering the order within the block as a block may contain more than one event.

Reply:
Is the suggestion in reference to `version_id`'s, [here](https://github.com/decentralized-identity/ethr-did-resolver/blob/master/doc/did-method-spec.md#versionid-query-string-parameter), the spec only takes into account the block number -- we could add an extra index for events, but maybe more complicated

Reply:
The json specification allows adding new entries, so, yes, we could also add the event index here as well.

Reply:
There is nothing that prevents multiple transactions in a single block for the did-ether contract.    In fact that would be easy to create.  So we should consider that block number is simply an index, not an exclusive id.   The way I handled this in the [reference] (https://github.com/jac18281828/didethresolver/blob/f89c757388aaa40c4e003b44273502a087ce871b/dideth/src/lib.rs#L121) was to walk in reverse order back though the events and check for revocation.      An expired validity is the same as being revoked.

Reply:
That's what is happening here --- all the events are crawled in `changelog`, reversed, and this code is walking through them + updating version id in the process + checking for validity/revocation in the ethr document builder methods in `ethr.rs`

Reply:
Okay, thanks this is what I was asking about then.